### PR TITLE
Retrieve articles

### PR DESCRIPTION
**Pull request by jmramos02**
/cc @zendesk/mintegrations @miogalang @joseconsador

### Description
Add retrieval of articles

### Dependencies
* None

### References
* JIRA: https://zendesk.atlassian.net/browse/MI-421

### Risks
* [medium] Helpdesk may not work

### CODE REVIEW

File: src/Zendesk/API/Resources/HelpCenter.php [hunk 1]
```diff
@@ -36,7 +37,8 @@ public function __construct(HttpClient $client)
     public static function getValidSubResources()
     {
         return [
-            'categories' => Categories::class
+            'categories'    => Categories::class,
+            'articles'      => Articles::class
```
Comment (miogalang) [inline src/Zendesk/API/Resources/HelpCenter.php]: add a , at the end so it looks nicer

File: src/Zendesk/API/Resources/HelpCenter/ResourceAbstract.php [hunk 2]
```diff
@@ -18,4 +18,31 @@ protected function getResourceNameFromClass()
 
         return 'help_center/' . $resourceName;
     }
+
```
Comment (miogalang) [inline src/Zendesk/API/Resources/HelpCenter/ResourceAbstract.php]: Is this duplicated code from the other abstract?

File: src/Zendesk/API/Traits/Resource/Localize.php [hunk 3]
```diff
@@ -0,0 +1,26 @@
+<?php
```
Comment (miogalang) [inline src/Zendesk/API/Traits/Resource/Localize.php]: I think the name should be `Locales` ? what do you think?

Comment (jmramos02) [inline src/Zendesk/API/Traits/Resource/Localize.php]: I think it is, renaming..

File: src/Zendesk/API/Resources/HelpCenter/ResourceAbstract.php (hunk 2 above)
Comment (jmramos02) [inline src/Zendesk/API/Resources/HelpCenter/ResourceAbstract.php]: Yes it is

Comment (miogalang) [inline src/Zendesk/API/Resources/HelpCenter/ResourceAbstract.php]: We should try not to replicate code, and keep it DRY. Maybe pull this out into a trait? do they share any other methods?

File: src/Zendesk/API/Resources/HelpCenter/ResourceAbstract.php [hunk 4]
```diff
@@ -18,4 +18,31 @@ protected function getResourceNameFromClass()
...
+    /**
+     * Generate a route depending on a localization set
+     * @param string $name
+     * @param array $params
+     */
+    public function getRoute($name, array $params = [])
+    {
+        $routesWithLocale = ['findAll', 'find', 'create', 'update'];
```
Comment (joseconsador) [inline src/Zendesk/API/Resources/HelpCenter/ResourceAbstract.php]: @jmramos02 What would happen if there are resources that don't use all of these?

Comment (jmramos02) [inline src/Zendesk/API/Resources/HelpCenter/ResourceAbstract.php]: Will pull it out on a trait so it can be optional

**Comment by jmramos02** [comment]
💇 @miogalang @joseconsador

File: src/Zendesk/API/Resources/HelpCenter/Articles.php [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+    /**
+     * @{inheritdoc}
+     */
+    protected $objectName = 'article';
+
+    /**
+     * @var locale
+     */
+    private $locale;
```
Comment (miogalang) [inline src/Zendesk/API/Resources/HelpCenter/Articles.php]: Please move this to the locales trait

**Comment by jmramos02** [comment]
💇 @miogalang

**Comment by miogalang** [comment]
👍 Good job

**Comment by joseconsador** [comment]
👍

**Pull request merged**